### DISCUSSION TITLE
uWS closes C# ClientWebSocket connection when any data is sent.

### DISCUSSION THREAD

**Discussion by sakidev** [uNetworking/uWebSockets.js#174] [Q&A] [answered]
Hey, I wanted to report this bug.

For reasons, I need this lib to be able to communicate with Microsoft's ClientWebSocket (C# .NET WebSocket client that you can see here: https://docs.microsoft.com/es-es/dotnet/api/system.net.websockets.clientwebsocket?view=netframework-4.8).

In the past, I used to work with your old library and everything went okay, but when trying out this new one, I find that ClientWebSocket can indeed establish a connection with uWebSockets but if I ever send data to it, uWebSockets will close the connection unexpectedly.

I am simply using this method to send data to uWebSockets async:
https://docs.microsoft.com/es-es/dotnet/api/system.net.websockets.clientwebsocket.sendasync?view=netframework-4.8#System_Net_WebSockets_ClientWebSocket_SendAsync_System_ArraySegment_System_Byte__System_Net_WebSockets_WebSocketMessageType_System_Boolean_System_Threading_CancellationToken_

Its use is very simple:

`ws.SendAsync(new ArraySegment<byte>(bytes), WebSocketMesssageType.Binary, true, _cancellationToken);`

And it simply makes the websocket connection crash.

This does not happen in your old lib. Why could it be? I was hoping to be able to use this new one but looks like I'll have to relay on the deprecated lib for now until we find the issue.

The Exception it throws is very generic:
![image](https://user-images.githubusercontent.com/49696473/62027600-58ab7b00-b1de-11e9-8d5f-88afb484db3b.png)

At first I thought this might be a Unity problem, but I also created a standalone C# Console Application and tried to do the same, the error it throws is also very generic, "The connection with the server finished abnormally".

Again, this only happens when uWS receives any data from ClientWebSocket. If I don't to anything other than connect or close the connection, nothing else happens.

**Comment by ghost** [comment, marked as answer]
Make sure the message size is less than 16kb or whatever maxPayloadLength you specified

**Comment by sakidev** [comment]
> Make sure the message size is less than 16kb or whatever maxPayloadLength you specified

yep it is, the message size is actually just a single byte (a packet header basically), for testing purposes.

**Comment by ghost** [comment]
I can't help you with this, you'll have to debug what happens

**Comment by yovanoc** [comment]
Speaking about that, if we put Infinity or something like that to deactivate this, performances are decreased ? What  is the need of this parameter?
